.github/workflows/linux_nightlies.yml

```diff
@@ -0,0 +1,46 @@
+name: Nightly Linux Unit Tests
+
+on:
+  schedule:
```

Comment:
These are what the nightlies look like. Before merge I'll remove the `on: pull_request` and update the `cron` to run at 3 am. I just added the `on: pull_request` to test it out. I'll also change the `ref` of the github action to say `main`.

So if this pr is merged, this is the expected behavior:

### On pushes to prs and merges to main run: 
- linux unit tests with latest versions python=3.8, core deps [True, False]. Measure coverage and upload to codecov.
- windows unit tests with python 3.8, core deps False. Don't measure coverage.
- linux unit tests with minimum versions with python 3.7, core deps False. Don't measure coverage.

### At 3am
- linux unit tests with latest versions with python [3.7, 3.9], core deps False. Don't measure coverage.
- windows unit tests with python [3.7, 3.9], core deps False. Don't measure coverage.
- Not running minimum dependency versions as a nightly because some of the minimum package versions are not available for 3.9! I think just testing minimum versions with 3.7 is fine (and that's what we do now).


Sound good? The total number of unit test checks would be 17. FYI @chukarsten @dsherry